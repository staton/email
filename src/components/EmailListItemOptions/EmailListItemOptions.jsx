import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import _ from 'lodash';
import Email from '../../models/email';
import EmailListItemOptionsButton from '../EmailListItemOptionsButton/EmailListItemOptionsButton';
import {MdDelete, MdFlag, MdMoreHoriz} from 'react-icons/md'

const propTypes = {
    email: PropTypes.instanceOf(Email).isRequired
};

const defaultProps = {
};

export class EmailListItemOptions extends Component {

    constructor() {
        super();

        this.handleDeleteButtonClicked = this.handleDeleteButtonClicked.bind(this);
        this.handleFlagButtonClicked = this.handleFlagButtonClicked.bind(this);
        this.handleMoreButtonClicked = this.handleMoreButtonClicked.bind(this);
    }

	render() {
		return (
            <div className="EmailListItemOptions">
                <EmailListItemOptionsButton 
                    className="EmailListItemOptionsButton__more"
                    content={<MdMoreHoriz />} 
                    onClick={this.handleMoreButtonClicked}
                />
                <EmailListItemOptionsButton 
                    className="EmailListItemOptionsButton__flag"
                    content={<MdFlag />} 
                    onClick={this.handleFlagButtonClicked}
                />
                <EmailListItemOptionsButton 
                    className="EmailListItemOptionsButton__delete"
                    content={<MdDelete />} 
                    onClick={this.handleDeleteButtonClicked}
                />
            </div>
		);
    }

    /**
     * Called when the user taps the Delete button.
     * @param {object} e The event.
     */
    handleDeleteButtonClicked(e) {
        console.log('DELETE clicked');
    }

    /**
     * Called when the user taps the Flag button.
     * @param {object} e The event.
     */
    handleFlagButtonClicked(e) {
        console.log('FLAG clicked');
    }

    /**
     * Called when the user taps the More button.
     * @param {object} e The event.
     */
    handleMoreButtonClicked(e) {
        console.log('MORE clicked');
    }
}

function mapStateToProps(store, ownProps) {
    return {
        email: ownProps.email
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
    },
    dispatch);
}

EmailListItemOptions.propTypes = propTypes;
EmailListItemOptions.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(EmailListItemOptions);