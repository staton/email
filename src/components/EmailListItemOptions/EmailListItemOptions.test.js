import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import {EmailListItemOptions} from './EmailListItemOptions';
import Email from '../../models/email';
import EmailFlags from '../../models/emailFlags';

describe('<EmailListItemOptions />', () => {

    const GET_EMAIL = () => new Email(
        0, 'bob', 'bob@example.com', ['s@gmail.com'], [], [], 'hello', 'preview', 
        new Date('2018-05-05T12:05:00Z'), null, new EmailFlags(false, false, false, false));

    it('renders correctly', () => {
        const wrapper = shallow(
            <EmailListItemOptions
                email={GET_EMAIL()}
            />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    
});