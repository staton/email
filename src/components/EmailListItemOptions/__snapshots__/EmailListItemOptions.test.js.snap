// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailListItemOptions /> renders correctly 1`] = `
<div
  className="EmailListItemOptions"
>
  <EmailListItemOptionsButton
    className="EmailListItemOptionsButton__more"
    content={<Unknown />}
    onClick={[Function]}
  />
  <EmailListItemOptionsButton
    className="EmailListItemOptionsButton__flag"
    content={<Unknown />}
    onClick={[Function]}
  />
  <EmailListItemOptionsButton
    className="EmailListItemOptionsButton__delete"
    content={<Unknown />}
    onClick={[Function]}
  />
</div>
`;
